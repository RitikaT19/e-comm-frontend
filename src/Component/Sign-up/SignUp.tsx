import React, { useState } from "react";
import Button from "../../common/Button/Button";
import { Textfield } from "../../common/Textfield/Textfield";
import "../../styles/sign-up.css";

interface Props {
  clickSignUpButton: any;
}
export const SignUp: React.FC<Props> = ({ clickSignUpButton }) => {
  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const [firstName, setFirstName] = useState<string>("");
  const [lastName, setLastName] = useState<string>("")

  const handleEmailChange = {};
  const handlePasswordChange = {};
  return (
    <div className="sign-up-main-div">
      <div className="textfield-div">
        <Textfield
          label="First name"
          placeholder="Enter first name"
          id="first_name"
          onChange={(e: any) => setFirstName(e.target.value)}
          value={firstName}
        />

        <Textfield
          label="Last name"
          placeholder="Enter last name"
          id="last_name"
          onChange={(e: any) => setLastName(e.target.value)}
          value={lastName}
        />
        <Textfield
          label="Email"
          placeholder="Enter email address"
          id="email"
          onChange={(e: any) => setEmail(e.target.value)}
          value={email}
        />

        <Textfield
          label="Password"
          type="password"
          placeholder="Enter password here"
          id="password"
          onChange={(e: any) => setPassword(e.target.value)}
          value={password}
        />
        <div className="sign_up_button_div">
          <Button
            value="Sign up"
            id="sign_up_button"
            handleClick={clickSignUpButton}
          />
        </div>
      </div>
    </div>
  );
};
