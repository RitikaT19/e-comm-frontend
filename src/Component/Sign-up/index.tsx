import React, { useContext } from "react";
import { SignUp } from "./SignUp";
import { LoadingContext } from "../../contexts/Loading";
import { SignUpContext } from "../../contexts/SignUp";
import { addUser, clearErrors } from "../../actions/signUp";

export const SignUpAdmin: React.FC = () => {
  const { state: signUpState, dispatch: signUpDispatch } =
    useContext(SignUpContext);
  const { dispatch: loadingDispatch } = useContext(LoadingContext);



  const clickSignUpButton = async (data: any) => {
    await addUser(data)(signUpDispatch, loadingDispatch).then(() => {
      console.log("add a new user", data);
    });
    // clearErrors(signUpDispatch)
  };
  return (
    <div>
      <SignUp handleSignUpButton={clickSignUpButton} 
      errorMessage = {signUpState.error}
      successMessage = {signUpState.addUserSuccess}/>
    </div>
  );
};
