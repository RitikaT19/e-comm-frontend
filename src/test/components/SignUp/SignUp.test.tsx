import React, { Component } from "react";
import { mount, ReactWrapper } from "enzyme";
import { SignUp } from "../../../component/Sign-up/SignUp";

describe("Testing SignUp Component", () => {
  // create a wrapper object
  let wrapper: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>>;

  const handleSignUpButton: any = jest.fn();
  const errorMessage = "";
  const successMessage = "";

  const defaultProps = {
    handleSignUpButton,
    errorMessage,
    successMessage,
  };

  beforeEach(() => {
    wrapper = mount(<SignUp {...defaultProps} />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should check error message if any field is empty", () => {
    wrapper.find("#sign_up_button_element").simulate("click");
    expect(wrapper.find("#sign-in-empty-field-error").text()).toEqual(
      "Please fill all the fields"
    );
  });

  it("should have 1 button component", () =>{
    expect(wrapper.find("Button")).toHaveLength(1)
})

it("should have 4 TextField component", () =>{
    let textFields = wrapper.find("TextField").length;
    expect(wrapper.find("TextField")).toHaveLength(textFields)
})
it("should call the handle change prop when a user types something in the email input field",()=>{
    wrapper.find("#sign_up_button_element").simulate("click")
    const event = {target: {value:"tests"}};
    const event2 = {target: {value: "test 123"}};
    wrapper.find("#email_element").simulate("change", event2)
    wrapper.find("#email_element").simulate("change", event)
})
it("should call the handle change prop when a user types something in the firstName input field",()=>{
    wrapper.find("#sign_up_button_element").simulate("click")
    const event = {target: {value:"tests"}};
    const event2 = {target: {value: "test 123"}};
    wrapper.find("#first_name_element").simulate("change", event2)
    wrapper.find("#first_name_element").simulate("change", event)
})
it("should call the handle change prop when a user types something in the firstName input field",()=>{
    wrapper.find("#sign_up_button_element").simulate("click")
    const event = {target: {value:"tests"}};
    const event2 = {target: {value: "test 123"}};
    wrapper.find("#last_name_element").simulate("change", event2)
    wrapper.find("#last_name_element").simulate("change", event)
})
it("should call the handle change prop when a user types something in the firstName input field",()=>{
    wrapper.find("#sign_up_button_element").simulate("click")
    const event = {target: {value:"tests"}};
    const event2 = {target: {value: "test 123"}};
    wrapper.find("#password_element").simulate("change", event2)
    wrapper.find("#password_element").simulate("change", event)
})
});
