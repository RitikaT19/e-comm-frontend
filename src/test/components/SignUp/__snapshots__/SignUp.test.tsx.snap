// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUp Component should match snapshot 1`] = `
<SignUp
  errorMessage=""
  handleSignUpButton={[MockFunction]}
  successMessage=""
>
  <div
    className="sign-up-main-div"
  >
    <img
      alt="background"
      className="background"
      src="background.jpg"
    />
    <div
      className="textfield-div"
    >
      <p
        className="sign-up-heading"
      >
        Make a new Admin!
      </p>
      <Textfield
        error=""
        id="first_name"
        label="First name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter first name"
        value=""
      >
        <div
          className="textfield-container"
        >
          <label
            htmlFor="input-field"
          >
            First name
          </label>
          <input
            className="textfield"
            id="first_name_element"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter first name"
            value=""
          />
        </div>
      </Textfield>
      <Textfield
        error=""
        id="last_name"
        label="Last name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter last name"
        value=""
      >
        <div
          className="textfield-container"
        >
          <label
            htmlFor="input-field"
          >
            Last name
          </label>
          <input
            className="textfield"
            id="last_name_element"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter last name"
            value=""
          />
        </div>
      </Textfield>
      <Textfield
        error=""
        id="email"
        label="Email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter email address"
        value=""
      >
        <div
          className="textfield-container"
        >
          <label
            htmlFor="input-field"
          >
            Email
          </label>
          <input
            className="textfield"
            id="email_element"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter email address"
            value=""
          />
        </div>
      </Textfield>
      <Textfield
        error=""
        id="password"
        label="Password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter password here"
        type="password"
        value=""
      >
        <div
          className="textfield-container"
        >
          <label
            htmlFor="input-field"
          >
            Password
          </label>
          <input
            className="textfield"
            id="password_element"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter password here"
            type="password"
            value=""
          />
        </div>
      </Textfield>
      <div
        className="sign_up_button_div"
      >
        <Button
          handleClick={[Function]}
          id="sign_up_button"
          value="Make Admin"
        >
          <button
            className="button"
            id="sign_up_button_element"
            onClick={[Function]}
          >
            Make Admin
          </button>
        </Button>
      </div>
    </div>
  </div>
</SignUp>
`;
